Gems/Atom/Tools/MaterialCanvas/Code/Source/MaterialCanvasApplication.cpp

```diff
@@ -218,4 +222,45 @@ namespace MaterialCanvas
     {
         return m_window.get();
     }
+
+    void MaterialCanvasApplication::ApplyShaderBuildSettings()
+    {
+        // If minimal shader build settings are enabled, copy a settings registry file stub into the user settings folder. This will
+        // override AP and shader build settings, disabling shaders and shader variant building for inactive platforms and RHI. Copying any
+        // of these settings files requires restarting the application and the asset processor for the changes to be picked up.
+        if (auto fileIO = AZ::IO::FileIOBase::GetInstance())
+        {
+            const AZ::IO::Path settingsPath(
+                AZStd::string::format("%s/user/Registry/user_minimal_shader_build.setreg", AZ::Utils::GetProjectPath().c_str()));
+            const AZ::IO::Path settingsPathStub(
+                AtomToolsFramework::GetPathWithoutAlias("@gemroot:MaterialCanvas@/Registry/user_minimal_shader_build.setregstub"));
+            const AZ::IO::Path settingsPathDx12(
+                AZStd::string::format("%s/user/Registry/user_minimal_shader_build_dx12.setreg", AZ::Utils::GetProjectPath().c_str()));
+            const AZ::IO::Path settingsPathDx12Stub(
+                AtomToolsFramework::GetPathWithoutAlias("@gemroot:MaterialCanvas@/Registry/user_minimal_shader_build_dx12.setregstub"));
```

Comment:
Appending paths shouldn't be done using string format but instead the Path append operator.
```suggestion
            const AZ::FixedMaxPath projectPath = AZ::Utils::GetProjectPath();
            const AZ::FixedMaxPath materialCanvasGemPath = AZ::Utils::GetGemPath("MaterialCanvas");
            const auto settingsPath(projectPath / AZ::SettingsRegistryInterface::DevUserRegistryFolder / "user_minimal_shader_build.setreg");
            const auto settingsPathStub(materialCanvasGemPath / AZ::SettingsRegistryInterface::RegistryFolder / "user_minimal_shader_build.setregstub");
            const auto settingsPathDx12(projectPath / AZ::SettingsRegistryInterface::DevUserRegistryFolder / "user_minimal_shader_build_dx12.setreg");
            const auto settingsPathDx12Stub(materialCanvasGemPath / AZ::SettingsRegistryInterface::RegistryFolder / "user_minimal_shader_build_dx12.setregstub"
```